dashboard/app/controllers/teacher_dashboard_controller.rb

```diff
@@ -1,6 +1,16 @@
 class TeacherDashboardController < ApplicationController
   load_and_authorize_resource :section
 
+  rescue_from CanCan::AccessDenied do
+    if params[:course_version_name]
+      redirect_to "/courses/#{params[:course_version_name]}"
+    elsif params[:unitName]
```

Comment:
We should probably match caps case for these.
```suggestion
    elsif params[:unit_name]
```